docs/SPEC.md

```diff
@@ -3685,31 +3685,39 @@ from(bucket: "telegraf/autogen")
-| org        | string                | Org is the organization name of the bucket.                                                                                                                                                                                        |
-| orgID      | string                | OrgID is the organization ID of the bucket.                                                                                                                                                                                        |
-| host       | string                | Host is the location of a remote host to write to. Defaults to `""`.                                                                                                                                                               |
-| token      | string                | Token is the authorization token to use when writing to a remote host. Defaults to `""`.                                                                                                                                           |
-| timeColumn | string                | TimeColumn is the name of the time column of the output.  Defaults to `"_time"`.                                                                                                                                                   |
-| tagColumns | []string              | TagColumns is a list of columns to be used as tags in the output. Defaults to all columns of type string, excluding all value columns and the `_field` column if present.                                                          |
-| fieldFn    | (r: record) -> record | Function that takes a record from the input table and returns an object. For each record from the input table `fieldFn` returns on object that maps output field key to output value. Default: `(r) => ({ [r._field]: r._value })` |
-
-TODO(nathanielc): The fieldFn is not valid and needs to change. It uses dynamic object keys which is not allowed.
+The `to` operation takes data from a stream and writes it to a bucket.
+`to` has the following properties:
+
+| Name              | Type           | Description                                                                                                                                                                                                                        |
+| ----              | ----           | -----------                                                                                                                                                                                                                        |
+| bucket            | string         | Bucket is the bucket name into which data will be written.                                                                                                                                                                         |
```

Comment:
`bucket` and `bucketID` are specified as mutually exclusive. The type system does not allow for this. Instead I propose we only have a single `bucket` parameter that can take either a bucket name or ID as input (they're both strings after all).

I say we do the same for `org` and `orgID`. Let's just have a single `org` parameter.

Reply:
How would you distinguish bucketIDs from bucket names? A bucket name that looks like a bucketID is a valid bucket name. 

I agree the mutual exclusiveness of the arguments is odd, but we can't just merge them, maybe there is another way?

Reply:
Perhaps you give precedence to bucketIDs. I.e you check if the provided argument is a valid bucketID and if not you check if it is a valid bucket name?

Reply:
That would mean that bucket names cannot be valid bucketIDs, which is not true.

Reply:
But is it possible for a user to have a bucket with name `mybucket` and another bucket whose ID is `mybucket`?  (I know `mybucket` is not a valid bucketID, but you get my point)

Another option is for `to` to only take bucket names exclusively (or bucketIDs exclusively).

Reply:
Yes it is possible to have a bucket with ID `03cbbf8eb3a9e000` and another bucket with name `03cbbf8eb3a9e000`. It would be a bad idea as a user to create such a named bucket but its not illegal. 

I just tried it out on my prod account to make sure, and yes it is possible.

Reply:
We could make it clear in the spec that we try bucketIDs first and only if that fails do we try bucket names?

Reply:
After discussing with the team, I will create a separate issue to address this.